source/state_representation/include/state_representation/Robot/JointState.hpp

```diff
@@ -260,13 +415,17 @@ inline void JointState::set_state_variable(Eigen::VectorXd& state_variable, cons
   this->set_state_variable(state_variable, Eigen::VectorXd::Map(new_value.data(), new_value.size()));
 }
 
+inline void JointState::set_all_state_variables(const Eigen::VectorXd& new_values) {
+  this->set_positions(new_values.segment(0, this->get_size()));
+  this->set_velocities(new_values.segment(this->get_size(), this->get_size()));
+  this->set_accelerations(new_values.segment(2 * this->get_size(), this->get_size()));
+  this->set_torques(new_values.segment(3 * this->get_size(), this->get_size()));
+}
+
 inline JointState& JointState::operator=(const JointState& state) {
   State::operator=(state);
```

Comment:
If there this is a compatibility check, I don't have a remark, otherwise there would be one needed